src/main.cpp

```diff
@@ -210,15 +210,21 @@ void mainLoop(void)
         } if (sysHalPowerButtonAsserted()) {
           LOG_PRINT(INFO, "Power button pushed, preparing to power off...");
           timerHalBegin(&powerTimer, POWEROFF_TIMEOUT, false);
-	  alarmHalRing(ALARM_HAL_CONSTANT);
+	  alarmSet(&powerOffAlarm);
           powerOffStep = POWEROFF_STEP1;
         }
         break;
       case POWEROFF_STEP1:
 	 LOG_PRINT(INFO, "POWEROFF_STEP1");
-        if (comm.powerOff || (timerHalRun(&powerTimer) == HAL_TIMEOUT)) {
-	  LOG_PRINT(INFO, "Changing to POWEROFF_STEP2");
-          powerOffStep = POWEROFF_STEP2;
+	 if (comm.powerOff) {
+	   if(timerHalRun(&powerTimer) == HAL_TIMEOUT) {
+	   LOG_PRINT(INFO, "Changing to POWEROFF_STEP2");
+	   powerOffStep = POWEROFF_STEP2;
+	   }
+	 } else if(timerHalRun(&powerTimer) == HAL_TIMEOUT) {
+	   LOG_PRINT(INFO, "Canceling  power off");
+	   alarmSuppress(&powerOffAlarm);
+	   powerOffStep = POWEROFF_STEP0;
```

Comment:
The timer was originally indented to enable "peaceful" software shutdown in the event of a UI hang. If we dont really care about this (since if the UI is stuck, theres always the hard power off), we can get rid of the timer and have the controller ignore the button entirely. That way, if the UI never issues a powerOff command, the controller will just ignore it, making the UI in control of the power off sequence initialization.

Reply:
The timer I added to turn off the alarm. The idea is as soon as the power button is pressed you don't continue pressing it for 12 seconds in error. This means that the sound should also come in the vent is in active ventilation mode.

Reply:
The previous implementation was implemented in accordance with the diagrams Dawn created, though I did add that timer to catch UI hangs when I shouldnt have according to these flows. If you want to change the behavior, you should probably make sure the flow is clear:
Flow if not ventilating:
https://docs.google.com/drawings/d/18HMhex4woUm92CJvpRbYs7WUHl_WF7-8fJD0X3kRqfE/edit
Flow if ventilating:
https://docs.google.com/drawings/d/1NMcsxk5UIIg3xlkM4I24ktSFHBFxSSZU0pOrE6gMK1k/edit